src/meta/processors/usersMan/AuthenticationProcessor.cpp

```diff
@@ -0,0 +1,271 @@
+    folly::SharedMutex::WriteHolder wHolder(LockUtils::userLock());
+    const auto& user = req.get_user();
+    auto ret = getUserId(user.get_account());
+    if (ret.ok()) {
+        if (req.get_missing_ok()) {
+            resp_.set_code(cpp2::ErrorCode::SUCCEEDED);
+        } else {
+            LOG(ERROR) << "Create User Failed :" << user.get_account() << " have existed";
+            resp_.set_code(cpp2::ErrorCode::E_EXISTED);
+        }
+        resp_.set_id(to(ret.value(), EntryType::USER));
+        onFinished();
+        return;
+    }
+
+    UserID userId = autoIncrementId();
+    std::vector<kvstore::KV> data;
+    data.emplace_back(MetaServiceUtils::indexUserKey(user.get_account()),
+                      std::string(reinterpret_cast<const char*>(&userId), sizeof(userId)));
+    LOG(INFO) << "Create User " << user.get_account() << ", userId " << userId;
+    data.emplace_back(MetaServiceUtils::userKey(userId),
+                      MetaServiceUtils::userVal(req.get_encoded_pwd(), user));
+    resp_.set_code(cpp2::ErrorCode::SUCCEEDED);
+    resp_.set_id(to(userId, EntryType::USER));
+    doPut(std::move(data));
+}
+
+
+void AlterUserProcessor::process(const cpp2::AlterUserReq& req) {
+    folly::SharedMutex::WriteHolder wHolder(LockUtils::userLock());
+    const auto& user = req.get_user_item();
+    auto ret = getUserId(user.get_account());
+    if (!ret.ok()) {
+        resp_.set_code(cpp2::ErrorCode::E_NOT_FOUND);
+        onFinished();
+        return;
+    }
+    UserID userId = ret.value();
+    auto userKey = MetaServiceUtils::userKey(userId);
+    std::string val;
+    auto result = kvstore_->get(kDefaultSpaceId, kDefaultPartId, userKey, &val);
+    if (result != kvstore::ResultCode::SUCCEEDED) {
+        resp_.set_code(cpp2::ErrorCode::E_NOT_FOUND);
+        onFinished();
+        return;
+    }
+    std::vector<kvstore::KV> data;
+    data.emplace_back(std::move(userKey),
+                      MetaServiceUtils::replaceUserVal(user, val));
+    resp_.set_code(cpp2::ErrorCode::SUCCEEDED);
+    resp_.set_id(to(userId, EntryType::USER));
+    doPut(std::move(data));
+}
+
+
+void DropUserProcessor::process(const cpp2::DropUserReq& req) {
+    folly::SharedMutex::WriteHolder wHolder(LockUtils::userLock());
+    auto ret = getUserId(req.get_account());
+    if (!ret.ok()) {
+        if (req.get_missing_ok()) {
+            resp_.set_code(cpp2::ErrorCode::SUCCEEDED);
+        } else {
+            LOG(ERROR) << "Drop User Failed :" << req.get_account() << " not found.";
+            resp_.set_code(cpp2::ErrorCode::E_NOT_FOUND);
+        }
+        onFinished();
+        return;
+    }
+    std::vector<std::string> keys;
+    keys.emplace_back(MetaServiceUtils::indexUserKey(req.get_account()));
+    keys.emplace_back(MetaServiceUtils::userKey(ret.value()));
+
+    // Collect related roles by user.
+    std::unique_ptr<kvstore::KVIterator> iter;
+    std::string prefix = "__roles__";
```

Comment:
I think the prefix should be used kRolesTable although it define in cpp file

Reply:
Good ! We are think alike, I don't feel comfortable using it here either. Let me try.